src/test/java/com/fittracker/fittracker/service/WeightServiceTest.java

```diff
@@ -65,67 +71,69 @@ static void AfterAll() {
-                    .isInstanceOf(WeightNotFoundException.class)
-                    .hasMessageContaining("Weight not found for date: 2023-10-10");
+        void givenDateFound_shouldThrowException() {
+            when(weightRepository.existsByDateAndUserId(any(),any())).thenReturn(true);
 
-            verify(weightRepository).findByDateAndUserId(LocalDate.of(2023, 10, 10), TEST_UUID);
+            assertThatThrownBy(()-> weightService.save(WEIGHT_REQUEST))
+                    .isInstanceOf(WeightAlreadyExistsException.class)
+                    .hasMessageContaining("Weight already exists for date: 2023-10-10");
+            verify(weightRepository).existsByDateAndUserId(TEST_DATE,TEST_UUID);
             verifyNoMoreInteractions(weightRepository);
         }
 
     }
 
     @Nested
-    class Save {
-
-
+    class FindByDate {
         @Test
-        void givenNoDateFound_shouldSaveAndReturnWeightResponse() {
-            when(weightRepository.existsByDateAndUserId(any(), any())).thenReturn(false);
-            when(weightRepository.save(any())).thenReturn(weight);
+        void givenDateFound_shouldReturnWeightResponse() {
+            when(weightRepository.findByDateAndUserId(any(), any())).thenReturn(of(weight));
 
-            var result = weightService.save(WEIGHT_REQUEST);
+            var result = weightService.findByDate(LocalDate.of(2023, 10, 10));
 
             assertThat(result).isEqualTo(WEIGHT_RESPONSE);
-            verify(weightRepository).existsByDateAndUserId(TEST_DATE, TEST_UUID);
-            verify(weightRepository).save(weightCaptor.capture());
-
-            assertThat(weightCaptor.getValue()).usingRecursiveComparison().isEqualTo(weight);
+            verify(weightRepository).findByDateAndUserId(LocalDate.of(2023, 10, 10), TEST_UUID);
             verifyNoMoreInteractions(weightRepository);
         }
 
         @Test
-        void givenDateFound_shouldThrowException() {
-            when(weightRepository.existsByDateAndUserId(any(), any())).thenReturn(true);
+        void givenNoDateFound_shouldThrowWeightNotFoundException() {
+            when(weightRepository.findByDateAndUserId(any(), any())).thenReturn(empty());
 
-            assertThatThrownBy(() -> weightService.save(WEIGHT_REQUEST))
-                    .isInstanceOf(WeightAlreadyExistsException.class)
-                    .hasMessageContaining("Weight already exists for date: 2023-10-10");
-            verify(weightRepository).existsByDateAndUserId(TEST_DATE, TEST_UUID);
+            assertThatThrownBy(() -> weightService.findByDate(LocalDate.of(2023, 10, 10)))
+                    .isInstanceOf(WeightNotFoundException.class)
+                    .hasMessageContaining("Weight not found for date: 2023-10-10");
+
+            verify(weightRepository).findByDateAndUserId(LocalDate.of(2023, 10, 10), TEST_UUID);
             verifyNoMoreInteractions(weightRepository);
         }
+
```

Comment:
please don't rearrange the existing code for no reason (if it's not purely refactor PR), it makes it difficult to see if any changes were made

Reply:
Sure. Hadn't thought about reviewer's point of view.

Reply:
please undo that change